Fireworks engine

1stprinciple opened the pull request.

1stprinciple committed:
init

1stprinciple committed:
init

1stprinciple committed:
self.hybrid_engine = False

1stprinciple committed:
hack for Role.ActorRollout

1stprinciple committed:
trainer.val_before_train=False \

1stprinciple committed:
lora

1stprinciple committed:
save_lora_adapter

1stprinciple committed:
print("lora_adapter_path"

1stprinciple committed:
\

1stprinciple committed:
_upload_model

1stprinciple committed:
rename

pgarbacki commented on rllm/engine/rollout/fireworks_engine.py:
```diff
@@ -0,0 +1,65 @@
+            try:
+                self.tool_parser = ToolParser.get_parser(self.tokenizer)
+            except Exception:
+                print(f"Warning: No tool parser found for {self.tokenizer.name_or_path}. Tool calls not be parsed.")
```
Tool calls will not be parsed.

pgarbacki commented on rllm/engine/rollout/fireworks_engine.py:
```diff
@@ -0,0 +1,65 @@
+
+    def update_model_weights(self, lora_adapter_path: dict):
+        print("updating fireworks deployment weights")
+        pass
```
hmm, is this correct?

1stprinciple replied:
Done.

pgarbacki commented on rllm/trainer/verl/agent_workflow_trainer_fireworks.py:
```diff
@@ -0,0 +1,492 @@
+
+    def init_workers(self):
+        assert not self.hybrid_engine, "Pipeline trainer does not support hybrid engine, assumes Rollout and Actor are not in the different worker group"
+        """Init resource pool and worker group"""
```
move it one line up

1stprinciple replied:
Done.

pgarbacki commented on rllm/trainer/verl/agent_workflow_trainer_fireworks.py:
```diff
@@ -0,0 +1,492 @@
+
+        from transformers import AutoTokenizer
+        fireworks_engine = FireworksEngine(
+            model="accounts/fireworks/models/qwen3-30b-a3b",
```
Should this be hard-coded?

1stprinciple replied:
Done.

pgarbacki commented on rllm/trainer/verl/agent_workflow_trainer_fireworks.py:
```diff
@@ -0,0 +1,492 @@
+        from transformers import AutoTokenizer
+        fireworks_engine = FireworksEngine(
+            model="accounts/fireworks/models/qwen3-30b-a3b",
+            tokenizer=AutoTokenizer.from_pretrained("Qwen/Qwen3-0.6B"),
```
Ditto.

1stprinciple replied:
Done

pgarbacki commented on rllm/trainer/verl/agent_workflow_trainer_fireworks.py:
```diff
@@ -0,0 +1,492 @@
+        """
+        The training loop of PPO. Adapted to train the underlying model of agent.
+        """
+        from verl.utils.tracking import Tracking
```
Move the imports to the top.

1stprinciple replied:
Done.

pgarbacki commented on rllm/trainer/verl/agent_workflow_trainer_fireworks.py:
```diff
@@ -0,0 +1,492 @@
+from verl.trainer.ppo.ray_trainer import (
+    AdvantageEstimator,
+    RayPPOTrainer,
+    RayWorkerGroup,
```
This is already imported in L15. Is it the same or a different class?

1stprinciple replied:
It's the same. Removed this one.

pgarbacki commented on rllm/trainer/verl/agent_workflow_trainer_fireworks.py:
```diff
@@ -0,0 +1,492 @@
+    apply_kl_penalty,
+    compute_advantage,
+    compute_data_metrics,
+    compute_throughout_metrics,
```
Is this spelled correctly?

1stprinciple replied:
Yes.
https://verl.readthedocs.io/en/latest/_modules/verl/trainer/ppo/ray_trainer.html

pgarbacki commented on rllm/trainer/verl/agent_workflow_trainer_fireworks.py:
```diff
@@ -0,0 +1,492 @@
+        self._load_checkpoint()
+
+        # perform validation before training
+        import time
```
This is already imported at the top.

1stprinciple replied:
Done.

1stprinciple committed:
finish fireworks_engine

1stprinciple committed:
30b

1stprinciple committed:
self.model = f"{self._account_id}/{fireworks_model_id}#{self._account_id}/{self._deployment_id}"

1stprinciple committed:
SyncDeployedModel.DEPLOYED

1stprinciple committed:
address comments

1stprinciple committed:
ulysses_sequence_parallel_size=8

1stprinciple committed:
32b

1stprinciple committed:
"""Init resource pool and worker group"""

1stprinciple committed:
6 gpus

1stprinciple committed:
=6

1stprinciple committed:
Revert "=6"

This reverts commit ceb89400a9dbb9e67bbf93a8b58da5d9f0c26efe.

1stprinciple committed:
Revert "6 gpus"

This reverts commit 1a5a6cbd4209f3dc7dcc80b1eb2349fcac49461f.

1stprinciple committed:
=4

1stprinciple committed:
Qwen3-32B

1stprinciple committed:
14b

1stprinciple committed:
4b

1stprinciple committed:
address comments

1stprinciple committed:
train_batch_size=64

1stprinciple committed:
remove self.tokenizer = tokenizer

1stprinciple committed:
duplicate import for     RayWorkerGroup,

1stprinciple committed:
hasattr

1stprinciple committed:
hasattr(self, "rollout_wg")

1stprinciple committed:
ulysses_sequence_parallel_size=1

1stprinciple committed:
8B

1stprinciple committed:
try 235b

1stprinciple committed:
n_training_gpus_per_node=4

1stprinciple committed:
tensor_model_parallel_size=4

1stprinciple committed:
tensor_model_parallel_size=4

1stprinciple committed:
param_offload=False

1stprinciple committed:
72b

1stprinciple committed:
n_training_gpus_per_node=8

1stprinciple committed:
n_training_gpus_per_node=8

1stprinciple committed:
back to 30b

1stprinciple committed:
back to 30b

1stprinciple committed:
3

1stprinciple committed:
3

1stprinciple committed:
32B

1stprinciple committed:
32B

1stprinciple committed:
rename to pipeline

1stprinciple committed:
back to 30b

1stprinciple committed:
back to 30b

1stprinciple committed:
attention only

1stprinciple committed:
attention only

1stprinciple committed:
train_workflow_pipeline

1stprinciple committed:
add hendrycks math

1stprinciple committed:
remove eval protocol

test-qwen-1p5b-1

train_fireworks_math

n_training_gpus_per_node

prepare

from rllm.engine.rollout.rollout_engine import ModelOutput

model_output

from rllm.engine.rollout.model_output import ModelOutput

Revert "from rllm.engine.rollout.model_output import ModelOutput"

This reverts commit fa844e416c419f865c3333bca7bdd09e058c1997.

Revert "model_output"

This reverts commit 5bb185a305e5450b7359d5fd3b2fab04549b64b1.

Revert "from rllm.engine.rollout.rollout_engine import ModelOutput"

This reverts commit 5ec4cf5b963054b330dcd75f99a9a50f27dcbbb2.

math_workflow_reward_fn

lora

print

4b

max_tokens

8b

Qwen/Qwen3-8B

16384

1stprinciple committed:
revert solver_judge

1stprinciple committed:
formatting

1stprinciple committed:
formatting

1stprinciple committed:
Add readme
